lib/integrations/awsoidc/list_security_groups.go

```diff
@@ -110,22 +150,52 @@ func ListSecurityGroups(ctx context.Context, clt ListSecurityGroupsClient, req L
 		return nil, trace.Wrap(err)
 	}
 
-	ret := &ListSecurityGroupsResponse{}
+	return &ListSecurityGroupsResponse{
+		NextToken:      aws.ToString(securityGroupsResp.NextToken),
+		SecurityGroups: convertAWSSecurityGroups(securityGroupsResp.SecurityGroups),
+	}, nil
+}
 
-	if aws.ToString(securityGroupsResp.NextToken) != "" {
-		ret.NextToken = *securityGroupsResp.NextToken
+func convertAWSSecurityGroups(awsSG []ec2Types.SecurityGroup) []SecurityGroup {
+	ret := make([]SecurityGroup, 0, len(awsSG))
+	for _, sg := range awsSG {
+		ret = append(ret, SecurityGroup{
+			Name:          aws.ToString(sg.GroupName),
+			ID:            aws.ToString(sg.GroupId),
+			InboundRules:  convertAWSIPPermissions(sg.IpPermissions),
+			OutboundRules: convertAWSIPPermissions(sg.IpPermissionsEgress),
+		})
 	}
 
-	ret.SecurityGroups = make([]SecurityGroup, 0, len(securityGroupsResp.SecurityGroups))
-	for _, sg := range securityGroupsResp.SecurityGroups {
-		sgName := aws.ToString(sg.GroupName)
-		sgID := aws.ToString(sg.GroupId)
+	return ret
+}
 
-		ret.SecurityGroups = append(ret.SecurityGroups, SecurityGroup{
-			Name: sgName,
-			ID:   sgID,
+func convertAWSIPPermissions(permissions []ec2Types.IpPermission) []SecurityGroupRule {
+	rules := make([]SecurityGroupRule, 0, len(permissions))
+	for _, permission := range permissions {
+		ipProtocol := allProtocols
+		if aws.ToString(permission.IpProtocol) != "-1" {
+			ipProtocol = aws.ToString(permission.IpProtocol)
+		}
```

Comment:
It's not obvious what `"-1"` is here or why this makes sense, a comment would be helpful.